File: 2022/day1.py
from aoc import input_as_lines

inp = input_as_lines("day1.txt")
cals = []

# PART 1
bucket = 0
for i in inp:
    if i == "":
        cals.append(bucket)
        bucket = 0
    else:
        bucket += int(i.rstrip('\n'))
print(max(cals))

# PART 2
print(sum(sorted(cals, reverse=True)[:3]))

File: 2022/day1.dart
import 'dart:math';

import 'aoc.dart';

void main() {
  var input = inputAsLines(1);
  List<int> cals = [];

  // Part 1
  int bucket = 0;
  for (String i in input) {
    if (i.isEmpty) {
      cals.add(bucket);
      bucket = 0;
    } else {
      bucket += int.parse(i);
    }
  }
  print(cals.reduce(max));

  // Part 2
  cals.sort((b, a) => a.compareTo(b));
  print(cals.sublist(0, 3).reduce((a, b) => a + b));
}
